desktop/core/src/desktop/loglistener.py

```diff
@@ -0,0 +1,214 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import logging
+import logging.config
+import os
+import pickle
+import re
+import select
+import signal
+import socket
+import stat
+import struct
+import sys
+import threading
+
+from io import StringIO as string_io
+from socketserver import ThreadingTCPServer, StreamRequestHandler
+#
+#   The following code implements a socket listener for on-the-fly
+#   reconfiguration of logging.
+#
+#   _listener holds the server object doing the listening
+_udslistener = None
+def udslisten(server_address='/tmp/hue.uds', verify=None):
+  """
+  Start up a socket server on the specified unix domain socket, and listen for new
+  configurations.
+  """
+  class ConfigStreamHandler(StreamRequestHandler):
+    """
+    Handler for a streaming logging request.
+    This basically logs the record using whatever logging policy is
+    configured locally.
+    """
+
+    def handle(self):
+      """
+      Handle multiple requests - each expected to be a 4-byte length,
+      followed by the LogRecord in pickle format. Logs the record
+      according to whatever policy is configured locally.
+      """
+      while True:
+        chunk = self.connection.recv(4)
+        if len(chunk) < 4:
+          break
+        slen = struct.unpack('>L', chunk)[0]
+        chunk = self.connection.recv(slen)
+        while len(chunk) < slen:
+          chunk = chunk + self.connection.recv(slen - len(chunk))
+        obj = self.unPickle(chunk)
+        record = logging.makeLogRecord(obj)
+        self.handleLogRecord(record)
+
+    def unPickle(self, data):
+      return pickle.loads(data)
+
+    def handleLogRecord(self, record):
+      # if a name is specified, we use the named logger rather than the one
+      # implied by the record.
+      if self.server.logname is not None:
+        name = self.server.logname
+      else:
+        name = record.name
+      logger = logging.getLogger(name)
+      logger.handle(record)
+
+  class ConfigSocketReceiver(ThreadingTCPServer):
+    """
+    A simple TCP socket-based logging config receiver.
+    """
+    request_queue_size = 1
+
+    def __init__(self, server_address='/tmp/hue.uds', handler=None,
+                 ready=None, verify=None):
+      ThreadingTCPServer.__init__(self, server_address, handler)
+      logging._acquireLock()
+      self.abort = 0
+      logging._releaseLock()
+      self.timeout = 1
+      self.ready = ready
+      self.verify = verify
+      self.logname = None
+      self.server_address = server_address
+
+    def server_bind(self):
+      # Create a UDS socket
+      self.socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+      self.socket.bind(self.server_address)
```

Comment:
What happens if a duplicate process starts up? Would the second one fail to bind and exit?

Reply:
If duplicate process starts up then it will starts logging to the Hue log file